Libraries/Azure.psm1

```diff
@@ -426,13 +426,25 @@ Function Invoke-AllResourceGroupDeployments($SetupTypeData, $CurrentTestData, $R
 			$DiskType = "Unmanaged"
 		}
 
+		if (!$global:password -and !$global:sshPrivateKey) {
+			Throw "Please set sshPrivateKey or linuxTestPassword."
+		}
 		$UseSpecializedImage = $CurrentTestData.SetupConfig.ImageType -contains "Specialized"
 		$IsWindowsOS = $CurrentTestData.SetupConfig.OSType -contains "Windows"
 		if ($IsWindowsOS) {
 			$OSType = "Windows"
+			if (!$global:password) {
+				Throw "Please provide password when provision a Windows VM."
+			}
+			Write-LogDbg "Windows OS, use password authentication, reset Key into empty."
+			$global:sshPrivateKey = ""
 		}
 		else {
 			$OSType = "Linux"
+			if ($global:sshPrivateKey) {
+				Write-LogDbg "Linux OS, use SSH key authentication, reset password into empty."
+				$global:password = ""
```

Comment:
Does it mean we can access to empty password to Linux VM? /etc/sshd/sshd.config can have different parameter in order to allow empty password access.

Reply:
no, when enable root, it also comes with ssh key, password auth is disabled.

Reply:
Right, that is true in general. Sometimes non-sense change comes out in the new image. I think "password auth is disabled" is at risk without checking in our test.